libraries/stdlib/src/kotlin/collections/Collections.kt

```diff
@@ -327,12 +327,12 @@ public inline fun <T, K : Comparable<K>> List<T>.binarySearchBy(
 
 /**
  * Searches this list or its range for an element for which [comparison] function returns zero using the binary search algorithm.
- * The list is expected to be sorted into ascending order according to the provided [comparison],
- * otherwise the result is undefined.
+ * The list is expected to be sorted so that the integer return values of the provided [comparison]
+ * form an ascending sequence on the list elements. Otherwise the result is undefined.
```

Comment:
"ascending sequence" that is sufficient but not necessary requirement. The necessary requirement is that the sign is ascending, i.e.  negative values come before zero and zeroes come before positive values.